x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/context/processors_context.tsx

```diff
@@ -153,9 +153,12 @@ export const PipelineProcessorsContextProvider: FunctionComponent<Props> = ({
           // These are the option names we get back from our UI
           const knownOptionNames = [
             ...Object.keys(processorTypeAndOptions.options),
+
             // We manually add fields that we **don't** want to be treated as "unknownOptions"
             'internal_networks',
             'internal_networks_field',
+            'transport',
+            'iana_number',
```

Comment:
I don't understand how that works and why we need to do that, would mind explaining this for me please?

Reply:
oh nice catch! This was needed before when we had a toggle that shows either one input or the other but now since they are both always on screen we don't need to whitelist these two.

We currently have some logic that help us not lose unknown fields in our processor (ie: if our UI hasn't been updated to support a new field, users can still create the processor with curl and the UI should not get rid of it even though its not showing it). This was introduced with https://github.com/elastic/kibana/pull/91146.

The problem that this bit of code solves is eagerly merging props coming in that are not returned by the form (ie: optionally rendered fields), so we can get around this by manually listing the fields we don't want merged back in. This was introduced with https://github.com/elastic/kibana/pull/103436.